streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -805,10 +805,10 @@ public void advanceWallClockTime(final Duration advance) {
 
     private Queue<ProducerRecord<byte[], byte[]>> getRecordsQueue(final String topicName) {
         final Queue<ProducerRecord<byte[], byte[]>> outputRecords = outputRecordsByTopic.get(topicName);
-        if (outputRecords == null) {
-            if (!processorTopology.sinkTopics().contains(topicName)) {
-                throw new IllegalArgumentException("Unknown topic: " + topicName);
-            }
+        if (outputRecords == null && !processorTopology.sinkTopics().contains(topicName)) {
```

Comment:
Seems good enough as a bug fix, but I was wondering whether we could detect the dynamic topic is configured or not to make sure we are not actually allowing some other bugs to catch in TTD

Reply:
What do you mean by "configured"? My understanding is that topics don't actually even "exist" in the TTD, we just use a MockProducer and grab output records from its history to enqueue in the `outputRecordsByTopic` map. So if an output topic hasn't been sent to yet, there's no way to say whether it does or does not exist

Reply:
We could detect if the processorTopology contains only `StaticTopicNameExtractors` and still throw in that case if the topic name isn't in the topology.

Reply:
Yeah, that's a good point. Took a quick look at this and personally I feel it's not worth it to add even more things for the internal topology to track. But it wouldn't really add any complexity so if this is a big concern I can go ahead and add in a flag to watch out for any non-static TopicNameExtractors

Reply:
Ok, let's just keep it in our back pocket for now.